mmhuman3d/core/cameras/camera_parameters.py

```diff
@@ -483,12 +487,45 @@ def __check_value_type__(self, key: Any, val: Any) -> None:
         Raises:
             TypeError:
                 Value is supported but doesn't match definition.
+
+        Returns:
+            val (Any): The value casted into correct format.
         """
-        if type(val) != self.__class__.SUPPORTED_KEYS[key]['type']:
+        np_type_mapping = {int: np.integer, float: np.floating}
+        supported_keys = self.__class__.SUPPORTED_KEYS
+        check_passed = False
+        ret_val = None
+        if supported_keys[key]['type'] == int or\
+                supported_keys[key]['type'] == float:
+            type_str = str(type(val))
+            if type(val) == self.__class__.SUPPORTED_KEYS[key]['type']:
+                check_passed = True
+                ret_val = val
+            if 'numpy' in type_str:
```

Comment:
why not use `isinstance(val, np.ndarray)` to check type?

Reply:
Because the type of val is something like np.int32 or np.float64, not confirmed before the key is determined. `isinstance(val, np.ndarray)` is False and `isinstance(val, np.int32)` is True.

Reply:
Not sure if I misunderstand, I did some experiments:
```python
>>> import numpy as np
>>> val = np.zeros(3, dtype=np.int32)
>>> isinstance(val, np.ndarray)
True
>>> val = np.zeros(3, dtype=np.float64)
>>> isinstance(val, np.ndarray)
True
```

Reply:
Actually users will assign a value to `Height` or `Width`, neither the whole array nor an array in shape [1, ].
```python
>>> np_array = np.zeros(3, dtype=np.int32)
>>> isinstance(np_array, np.ndarray)
True
>>> isinstance(np_array[0], np.ndarray)
False
>>> isinstance(np_array[0], np.int32)
True
>>> isinstance(np_array[0], np.int16)
False
>>> isinstance(np_array[0:1], np.ndarray)
True
```

Reply:
Great. I ignored that `val` is interally only a value.

Reply:
If this line only expects a value, can we use somehting like `np.integer` or `np.floating`?

```python
>>> a = np.zeros(3, dtype=np.int32)
>>> isinstance(a[0], np.integer)
True
>>> a = np.zeros(3, dtype=np.int64)
>>> isinstance(a[0], np.integer)
True
```

Reply:
For some rare cases, a user may define a class like `mynumpy` and will pass the first check. We may try whether there is a better way to check `numpy` and `torch`, rather than string. But it works for me if there is no better solutions.

Reply:
Literally `isinstance` is the same as `np.issubdtype`, the only difference is the first argument.
`isinstance`: Return whether an object is an instance of a class or of a subclass thereof.   
`np.issubdtype`: Returns True if first argument is a typecode lower/equal in type hierarchy.
Considering that we are dealing with numpy objects, `np.issubdtype` is safer.

Reply:
`inspect.getmodule()` does the job but it runs more slowly than `type()`. In 4056412, we use `split()` together `startswith()` to get the module name from string, avoiding the problem above.

Reply:
Looks great.